atom/browser/api/atom_api_menu.cc

```diff
@@ -23,9 +23,13 @@ Menu::Menu(v8::Isolate* isolate, v8::Local<v8::Object> wrapper)
     : model_(new AtomMenuModel(this)),
       parent_(nullptr) {
   InitWith(isolate, wrapper);
+  model_->AddObserver(this);
 }
 
 Menu::~Menu() {
+  if (model_) {
```

Comment:
Why do you need this check? Can the `model_` be destroyed before the `Menu` destructor is called?

Reply:
yes, exactly; `model_` can be destroyed before attempting to `RemoveObserver(this)` which would cause a crash so the check is needed